src/confd/src/infix-services.c

```diff
@@ -299,13 +302,71 @@ static int mdns_change(sr_session_ctx_t *session, uint32_t sub_id, const char *m
 	svc_enadis(ena, none, "avahi");
 	mdns_cname(session);
 
-	return put(cfg, srv);
+	return put(cfg);
 }
 
 static int lldp_change(sr_session_ctx_t *session, uint32_t sub_id, const char *module,
 	const char *xpath, sr_event_t event, unsigned request_id, void *_confd)
 {
-	return svc_change(session, event, xpath, "lldp", "lldpd");
+	struct lyd_node *node = NULL;
+	sr_data_t *cfg;
+	struct lyd_node *subnode;
+
+	switch (event) {
+	case SR_EV_ENABLED:
+	case SR_EV_CHANGE:
+		if (sr_get_data(session, xpath, 0, 0, 0, &cfg) || !cfg)
+			break;
+
+		node = cfg->tree;
+		if (lydx_is_enabled(node, "enabled")){
+			const char *tx_interval = lydx_get_cattr(node, "message-tx-interval");
+			FILE *fp = fopen(LLDP_CONFIG_NEXT, "w");
+			if (!fp) {
+				ERROR("Failed to open %s for writing", LLDP_CONFIG_NEXT);
+				break;
+			}
+			fprintf(fp, "configure lldp tx-interval %s\n", tx_interval);
+
+			LY_LIST_FOR(lyd_child(node), subnode) {
+    				if (!strcmp(subnode->schema->name, "port")) {
+					const char *port_name = lydx_get_cattr(subnode, "name");
+					const char *admin_status = lydx_get_cattr(subnode, "admin-status");
+
+					if (strcmp(admin_status, "tx-and-rx") == 0)
+						admin_status = "rx-and-tx";
+
+					fprintf(fp, "configure ports %s lldp status %s\n", port_name, admin_status);
+    				}
+			}
+			fclose(fp);
+		}
+
+		return put(cfg);
+
+	case SR_EV_DONE:
+		if (fexist(LLDP_CONFIG_NEXT)){
+			if (erase(LLDP_CONFIG))
+				ERROR("Failed to remove old %s", LLDP_CONFIG);
+
+			rename(LLDP_CONFIG_NEXT, LLDP_CONFIG);
+		}
+		else
+			if (erase(LLDP_CONFIG))
+				ERROR("Failed to remove old %s", LLDP_CONFIG);
+		
+		svc_change(session, event, xpath, "lldp", "lldpd");
+		break;
+
+	case SR_EV_ABORT:
+		erase(LLDP_CONFIG_NEXT);
```

Comment:
This is fine to ignore the result of, like you've done, since it's in our preparation path.